src/databases/Silo/avtSiloFileFormat.C

```diff
@@ -14491,19 +14491,50 @@ avtSiloFileFormat::GetExternalFacelist(int dom, const char *mesh)
 //    Moved debug5 output lines indicating that global node ids are being read
 //    down so that it appears in debug logs only when there is actually data
 //    being read and returned from the plugin.
-// ****************************************************************************
+//
+//    Mark C. Miller, Mon Feb 12 20:26:21 PST 2024
+//    Refactored away GetGlobalZoneIds and GetGlobalNodeIds into this one
+//    method that does both and does so for both UCD and POINT meshes.
+//    The main tricks are to a) set Silo's data read mask appropriately (and
+//    then also reset it when done) and then b) to allow VisIt's cache to take
+//    ownership of the resulting data. We do this by zeroing the Silo object's
+//    data member holding the global ids and keeping a separate local pointer
+//    to the data. For point meshes, nodes and zones are one in the same so the
+//    logic for both is the same. For UCD meshes, there is separate logic for
+//    global node ids and global zone ids embodied by a different set of args
+//    to the LOAD_GLOBAL_IDS macro'd logic.
+// ****************************************************************************
+#define LOAD_GLOBAL_IDS(DBTYP, GETFUNC, CKMEMBR, FREEFUNC, MASK, GIDTYP, NIDS, IDS) \
+do {                                                                       \
+    DBSetDataReadMask2(MASK);                                              \
+    DBTYP *obj = GETFUNC(domain_file, directory_mesh.c_str());             \
+    if (obj && obj->CKMEMBR)                                               \
+    {                                                                      \
+        gids = obj->IDS;                                                   \
+        ngids = obj->NIDS;                                                 \
+        gidtype = obj->GIDTYP;                                             \
+        obj->IDS = 0;                                                      \
+    }                                                                      \
+    FREEFUNC(obj);                                                         \
+} while (0)
 
 vtkDataArray *
-avtSiloFileFormat::GetGlobalNodeIds(int dom, const char *mesh)
+avtSiloFileFormat::GetGlobalIds(int dom, char const *mesh, char const *idtype)
 {
     DBfile *dbfile = GetFile(tocIndex);
 
-    char *meshname = AllocAndDetermineMeshnameForUcdmesh(dom, mesh);
+    int mtype = DB_UCDMESH;
+    char *meshname = AllocAndDetermineMeshnameForMeshType(dom, mesh, mtype);
```

Comment:
can we just use std::string here so we don't have to track and dealloc (which can be error prone)

Reply:
(i realize that is refactoring an existing method, maybe thats too much to ask)

Reply:
Done in 11ee1ea31fe2b35de868d2de675b4426b4c47649